### PR TITLE
Reliabletests

### PR DESCRIPTION
**Pull request by tmihalac**

### CODE REVIEW

File: mercury-api/src/main/java/com/redhat/mercury/operator/model/AbstractResourceStatus.java [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-import java.util.Optional;
-
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
 
 import io.fabric8.kubernetes.api.model.Condition;
 import io.sundr.builder.annotations.Buildable;
 import io.sundr.builder.annotations.BuildableReference;
 
+import java.util.ArrayList;
```
Comment (ruromero) [inline mercury-api/src/main/java/com/redhat/mercury/operator/model/AbstractResourceStatus.java]: java.util goes at the beginning

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/AbstractController.java [hunk 2]
```diff
@@ -23,6 +19,10 @@
 import io.javaoperatorsdk.operator.processing.event.source.EventSource;
 import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
 
+import java.util.List;
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/AbstractController.java]: same, fix the import order

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java [hunk 3]
```diff
@@ -38,6 +33,11 @@
 import io.strimzi.api.kafka.model.storage.SingleVolumeStorage;
 import io.strimzi.api.kafka.model.storage.Storage;
 
+import java.util.List;
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java]: Import order

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java [hunk 4]
```diff
@@ -68,12 +68,28 @@
...
             }
-            return updateStatusWithKafkaBrokerUrl(sdc);
+
+            control = updateStatusWithKafkaBrokerUrl(sdc);
+            if (control.isUpdateStatus()) {
+                return control;
+            }
+
+            if(areAllConditionsReady(sdc)){
+                control = updateStatusWithCondition(sdc, buildReadyCondition(CONDITION_READY));
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java]: maybe you can just return the control without checking `control.isUpdateStatus()`

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java [hunk 5]
```diff
@@ -48,28 +40,32 @@
...
-import io.strimzi.api.kafka.model.AclRuleTopicResourceBuilder;
 import io.strimzi.api.kafka.model.KafkaTopic;
 import io.strimzi.api.kafka.model.KafkaTopicBuilder;
-import io.strimzi.api.kafka.model.KafkaUser;
-import io.strimzi.api.kafka.model.KafkaUserAuthorizationSimpleBuilder;
-import io.strimzi.api.kafka.model.KafkaUserBuilder;
-import io.strimzi.api.kafka.model.KafkaUserTlsClientAuthenticationBuilder;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java]: import order

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java [hunk 6]
```diff
@@ -92,7 +88,7 @@
     private static final int SERVICE_PORT = 9000;
     private static final String COMMENT_LINE_REGEX = "(?m)^#.*";
     private static final String APP_LABEL_BIAN_PREFIX = "bian-";
-    private static final String OPENAPI_CM_SUFFIX = "-openapi";
+    public static final String OPENAPI_CM_SUFFIX = "-openapi";
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java]: does it need to be public?

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java [hunk 7]
```diff
@@ -139,69 +135,115 @@
...
+            return updateStatusWithCondition(sd, new ConditionBuilder()
+                    .withType(CONDITION_SERVICE_DOMAIN_CLUSTER_READY)
+                    .withStatus(STATUS_FALSE)
+                    .withReason(REASON_SDC)
+                    .withMessage(sdcName + " " + MESSAGE_SDC_NOT_FOUND)
+                    .build());
         }
 
-        if (sdc.getStatus().getCondition(CONDITION_READY) == null || Boolean.FALSE.toString().equalsIgnoreCase(sdc.getStatus().getCondition(CONDITION_READY).getStatus())) {
+        if (sdc.getStatus() != null && sdc.getStatus().getCondition(CONDITION_READY) == null || Boolean.FALSE.toString().equalsIgnoreCase(sdc.getStatus().getCondition(CONDITION_READY).getStatus())) {
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java]: the status cannot be null

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java [hunk 8]
```diff
@@ -139,69 +135,115 @@
...
             if (sd.getSpec().getExpose() != null && sd.getSpec().getExpose().contains(ServiceDomainSpec.ExposeType.http)) {
-                setStatusCondition(sd, CONDITION_INTEGRATION_READY, REASON_INTEGRATION, MESSAGE_INTEGRATION_NOT_READY, Boolean.FALSE);
                 final String sdConfigMapName = "integration-" + toLowerHyphen(sd.getSpec().getType().toString()) + "-http";
                 ConfigMap sdConfigMap = client.configMaps().inNamespace(client.getNamespace()).withName(sdConfigMapName).get();
+                final ServiceDomainSpec.Type sdType = sd.getSpec().getType();
+                final String sdTypeAsString = toLowerHyphen(sdType.toString());
 
-                String validationError = validateSdConfigMap(sd, sdConfigMapName, sdConfigMap);
-                if (validationError != null) {
+                if (sdConfigMap == null) {
```
Comment (ruromero) [inline mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java:179]: All the integration checks should be part of the createOrUpdateIntegration, create submethods if necessary

**Review by ruromero — CHANGES_REQUESTED** [review]
Just a few remarks but looks good overall, thanks.

File: mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java (hunk 6 above)
Comment (tmihalac) [reply mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java]: Using it in the test

**Review by ruromero — APPROVED** [review]

**Pull request merged**